packages/components/src/shortcuts/Shortcut.ts

```diff
@@ -0,0 +1,324 @@
+/* eslint-disable no-underscore-dangle */
+import React from 'react';
+import { ContextActionUtils } from '../context-actions';
+
+export enum MODIFIER {
+  CTRL = 'MODIFIER_CTRL',
+  CMD = 'MODIFIER_CMD',
+  ALT = 'MODIFIER_ALT',
+  OPTION = 'MODIFIER_OPTION',
+  SHIFT = 'MODIFIER_SHIFT',
+}
+
+// The value should match the KeyboardEvent.key value
+// This is the value which will be displayed if no display overrides are done in the OS specific getDisplayString methods
+export enum KEY {
+  A = 'A',
+  B = 'B',
+  C = 'C',
+  D = 'D',
+  E = 'E',
+  F = 'F',
+  G = 'G',
+  H = 'H',
+  I = 'I',
+  J = 'J',
+  K = 'K',
+  L = 'L',
+  M = 'M',
+  N = 'N',
+  O = 'O',
+  P = 'P',
+  Q = 'Q',
+  R = 'R',
+  S = 'S',
+  T = 'T',
+  U = 'U',
+  V = 'V',
+  W = 'W',
+  X = 'X',
+  Y = 'Y',
+  Z = 'Z',
+  BACKSPACE = 'Backspace',
+  ESCAPE = 'Escape',
+  ENTER = 'Enter',
+  DELETE = 'Delete',
+  SLASH = '/',
+}
+
+type ShortcutKeys = [...MODIFIER[], KEY];
```

Comment:
Neat. I sort of expected the syntax to be `[...MODIFIER, KEY]` (without the array on MODIFIER), but I guess I'm wrong

Reply:
Since `MODIFIER` is an enum (and thus the type refers to any value in the enum), you can't spread it. Specifically TS says "a rest element must be an array"

This mostly only helps with defining the shortcuts, but in 99% of cases we will probably be defining directly in the constructor params. The other 1% (tests mostly) you can declare the type I believe. It treats it as `(MODIFIER | KEY)[]` which is not compatible directly with this type